### ISSUE TITLE
Support for database notification

### ISSUE THREAD

**Issue by NobodyXu** [SOCI/soci#831]
Several database supported by `SOCI`, including
 - postgresql has [`Notify`](https://www.postgresql.org/docs/10/sql-notify.html)
 - sqlite3 has [`sqlite3_update_hook`](https://www.sqlite.org/c3ref/update_hook.html)
 - firebirds has [event system](https://www.firebirdsql.org/file/documentation/papers_presentations/Power_Firebird_events.pdf)

It is useful to clients who are interested in particular tables and need to keep the information on that table up-to-date.

**Comment by vadz** [comment]
This would be definitely very nice to have, although we'd still need to provide a polling-based generic fallback to make it usable with other databases.

Unfortunately I don't foresee having time to work on this myself in the near future, but any contributions would be very welcome.

**Comment by NobodyXu** [comment]
I am not familiar with SOCI, but I will try understanding it and add a callback when I am free.

**Comment by TwentyPast4** [comment]
I have implemented this and a few other firebird features for my own needs, feel free to use it!
https://github.com/IMAXAGaming/soci
contains:

- firebird transaction flags
- commit-retain and rollback-retain
- firebird events

**Comment by vadz** [comment]
Thanks, it's definitely a good start, but we need to at least check if this API is implementable with the other systems mentioned above.

And, of course, we need docs and tests if this is to be integrate into the official SOCI.

**Comment by TwentyPast4** [comment]
> Thanks, it's definitely a good start, but we need to at least check if this API is implementable with the other systems mentioned above.
> 
> And, of course, we need docs and tests if this is to be integrate into the official SOCI.

Yeah definitely. I've looked around a bit to see if some of this could be generalized, and I think it could.
I sadly don't have any time I could take to do this, so I suggest that if wanted, someone take this and work it into something more appropriate for a PR.

Great work on this lib though! ✌️
